### ISSUE TITLE
TensorRT engines can't be loaded in Triton, which created from demo/Diffusion

### ISSUE THREAD

**Issue by jslouisyou** [NVIDIA/TensorRT#2952]
## Description
Hello,
I'm newbie in TensorRT, unet and following Stable Diffusion demo, which describes in demo/Diffusion (https://github.com/NVIDIA/TensorRT/tree/release/8.6/demo/Diffusion) and I can export clip_encoder, unet, vae TensorRT engines as well and Stable Diffusion is also working well.

I created these models by following commands (`python3 demo_txt2img.py "Tall snowy mountain range, realistic, detailed, award winning photo, sunset, 4k" --width 768 --height 768 --scheduler DDIM --force-onnx-export --force-engine-build --build-preview-features --nvtx-profile --hf-token=${HF_TOKEN}`) and I didn't use any `config.pbtxt` since Triton Inference Server provides config automatically.

And all engines are built well:
```
1) clip
[I]     Configuring with profiles: [Profile().add('input_ids', min=(1, 77), opt=(1, 77), max=(4, 77))]
[I] Building engine with configuration:
    Flags                  | [FP16]
    Engine Capability      | EngineCapability.DEFAULT
    Memory Pools           | [WORKSPACE: 76556.81 MiB, TACTIC_DRAM: 81069.75 MiB]
    Tactic Sources         | []
    Profiling Verbosity    | ProfilingVerbosity.DETAILED
    Preview Features       | [FASTER_DYNAMIC_SHAPES_0805, DISABLE_EXTERNAL_TACTIC_SOURCES_FOR_CORE_0805]
[W] TensorRT encountered issues when converting weights between types and that could affect accuracy.
[W] If this is not the desired behavior, please modify the weights or retrain with regularization to adjust the magnitude of the weights.
[W] Check verbose logs for the list of affected weights.
[W] - 225 weights are affected by this issue: Detected subnormal FP16 values.
[W] - 1 weights are affected by this issue: Detected values less than smallest positive FP16 subnormal value and converted them to the FP16 minimum subnormalized value.
[W] - 1 weights are affected by this issue: Detected finite FP32 values which would overflow in FP16 and converted them to the closest finite FP16 value.
[I] Finished engine building in 72.574 seconds
[I] Saving engine to engine/clip.plan

2) unet
[I] Saving engine to engine/unet.plan
[I]     Configuring with profiles: [Profile().add('sample', min=(2, 4, 96, 96), opt=(2, 4, 96, 96), max=(8, 4, 96, 96)).add('encoder_hidden_states', min=(2, 77, 1024), opt=(2, 77, 1024), max=(8, 77, 1024)).add('timestep', min=[1], opt=[1], max=[1])]
[I] Building engine with configuration:
    Flags                  | [FP16]
    Engine Capability      | EngineCapability.DEFAULT
    Memory Pools           | [WORKSPACE: 76556.81 MiB, TACTIC_DRAM: 81069.75 MiB]
    Tactic Sources         | []
    Profiling Verbosity    | ProfilingVerbosity.DETAILED
    Preview Features       | [FASTER_DYNAMIC_SHAPES_0805, DISABLE_EXTERNAL_TACTIC_SOURCES_FOR_CORE_0805]
[W] - 272 weights are affected by this issue: Detected subnormal FP16 values.
[I] Finished engine building in 911.746 seconds
[I] Saving engine to engine/unet.plan

3) vae
Building TensorRT engine for onnx/vae.opt.onnx: engine/vae.plan
[I]     Configuring with profiles: [Profile().add('latent', min=(1, 4, 96, 96), opt=(1, 4, 96, 96), max=(4, 4, 96, 96))]
[I] Building engine with configuration:
    Flags                  | [FP16]
    Engine Capability      | EngineCapability.DEFAULT
    Memory Pools           | [WORKSPACE: 76556.81 MiB, TACTIC_DRAM: 81069.75 MiB]
    Tactic Sources         | []
    Profiling Verbosity    | ProfilingVerbosity.DETAILED
    Preview Features       | [FASTER_DYNAMIC_SHAPES_0805, DISABLE_EXTERNAL_TACTIC_SOURCES_FOR_CORE_0805]
operation.cpp:1494: DCHECK(cask_heuristic_result_.kernel_handle != fallback_cask_heuristic_result_.kernel_handle) failed. Invalid kernel handles.
[W] - 4 weights are affected by this issue: Detected subnormal FP16 values.
[I] Finished engine building in 330.707 seconds
[I] Saving engine to engine/vae.plan
```

But I faced some issues when I tried to use these engines in Triton. clip_encoder and vae engines were succefully loaded in Triton, but unet was not.
Here's the error message in Triton and model_repository structure is below:

```
I0510 02:43:01.111452 27564 server.cc:653] 
+----------------+---------+-------------------------------------------------------------------------------------------------------+
| Model          | Version | Status                                                                                                |
+----------------+---------+-------------------------------------------------------------------------------------------------------+
| clip_tokenizer | 4       | READY                                                                                                 |
| unet           | 4       | UNAVAILABLE: Internal: trt failed to set binding dimension to [1,4,96,96] for input 'sample' for unet |
| vae            | 4       | READY                                                                                                 |
+----------------+---------+-------------------------------------------------------------------------------------------------------+
```
```
$ tree
├── client.py
├── clip_tokenizer
│   ├── 1
│   └── model.plan
├── unet
│   ├── 1
│   └── model.plan
└── vae
    ├── 1
    └── model.plan
```

I digged into codes and I found that batch size is multiplied by 2.
```
class UNet(BaseModel):
.....
    def get_dynamic_axes(self):
        return {
            'sample': {0: '2B', 2: 'H', 3: 'W'},
            'encoder_hidden_states': {0: '2B'},
            'latent': {0: '2B', 2: 'H', 3: 'W'}
        }
.....
    def get_shape_dict(self, batch_size, image_height, image_width):
        latent_height, latent_width = self.check_dims(batch_size, image_height, image_width)
        return {
            'sample': (2*batch_size, self.unet_dim, latent_height, latent_width),
            'encoder_hidden_states': (2*batch_size, self.text_maxlen, self.embedding_dim),
            'latent': (2*batch_size, 4, latent_height, latent_width)
        }
.....
```

Here's my questions:
1. Are these engines not supported `dynamic batching` in Triton?
2. In this case, How can I load these TensorRT engine into Triton without modifying codes?

If you have any ideas, please let me know. It will be very helpful to me.
Thanks!

## Environment

**TensorRT Version**: `8.6.1.8`

**NVIDIA GPU**: `A100-SXM-80GB`

**NVIDIA Driver Version**: `510.73.08`

**CUDA Version**: `11.6`

**CUDNN Version**: `Cuda compilation tools, release 10.1, V10.1.243`

Operating System:

Python Version (if applicable): `Python 3.8.10`

Tensorflow Version (if applicable):

PyTorch Version (if applicable): `1.14.0a0+44dac51`

Baremetal or Container (if so, version):
`nvcr.io/nvidia/pytorch:23.02-py3, nvcr.io/nvidia/tritonserver:23.04-py3`

## Relevant Files

**Model link**:
https://github.com/NVIDIA/TensorRT/tree/release/8.6/demo/Diffusion

## Steps To Reproduce
1. Build TensorRT engine via command
2. Create model_repository folder and copy these engines
4. Run Triton Inference Server

**Commands or scripts**:
`tritonserver --model-repository=/models --log-verbose=1`

**Have you tried [the latest release](https://developer.nvidia.com/tensorrt)?**:

**Can this model run on other frameworks?** For example run ONNX model with ONNXRuntime (`polygraphy run <model.onnx> --onnxrt`):

**Comment by zerollzeng** [comment]
I'm sorry we are not able to help here, I would suggest asking this in the Nvidia Triton developer forum or corresponding repo if have.

**Comment by ttyio** [comment]
closing since no activity for more than 3 weeks, thanks!

**Comment by xizi** [comment]
same problem
